crates/paas/src/persistence.rs

```diff
@@ -0,0 +1,54 @@
+//! Persistence traits for task tracking
+
+use std::time::Instant;
+
+use crate::{
+    error::ProverServiceResult,
+    program::ProgramType,
+    task::{TaskId, TaskStatus},
+};
+
+/// Task metadata for persistence
+#[derive(Debug, Clone)]
+pub struct TaskRecord<T>
+where
+    T: Clone + Eq + std::hash::Hash + std::fmt::Debug + Send + Sync + 'static,
+{
+    pub task_id: T,
+    pub uuid: String,
+    pub status: TaskStatus,
+    pub created_at: Instant,
+    pub updated_at: Instant,
```

Comment:
Can probably make these private?

Similar comment as previously, this is a *lot* of type bounds to carry around that we could probably consolidate, *or* just make task IDs not generic.